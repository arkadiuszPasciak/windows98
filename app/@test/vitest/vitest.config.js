import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
