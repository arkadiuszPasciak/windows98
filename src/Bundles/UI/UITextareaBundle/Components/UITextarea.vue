<template>
  <div class="UITextarea" :class="classes">
    <label
      v-if="labelName"
      class="label"
      :data-test="`ui-textarea-label-${id}`"
      :for="id"
    >
      {{ labelName }}
    </label>
    <textarea
      :id="id"
      class="textarea"
      :value="modelValue"
      :disabled="disabled"
      :readonly="readonly"
      :data-test="`ui-textarea-${id}`"
      @input="
        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
      "
    />
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  const props = defineProps({
    type: {
      type: String as PropType<'basic' | 'notepad'>,
      default: 'basic',
    },
    id: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
    modelValue: {
      type: String,
      required: true,
    },
    labelName: {
      type: String,
      required: true,
    },
    labelPosition: {
      type: String as PropType<'top' | 'left'>,
      default: 'top',
    },
  })

  defineEmits(['update:modelValue'])

  const classes = [
    `label-position-${props.labelPosition}`,
    `${props.type === 'notepad' ? 'type-notepad' : ''}`,
  ] as Array<string>
</script>

<style
  lang="scss"
  scoped
  src="@Bundles/UITextareaBundle/Styles/UITextarea.styles.scss"
/>
