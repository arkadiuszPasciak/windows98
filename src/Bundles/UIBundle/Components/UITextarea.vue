<template>
  <div class="UITextarea" :class="classes">
    <label class="label" :for="id">{{ labelName }}</label>
    <textarea
      :id="id"
      class="textarea"
      :value="modelValue"
      :disabled="disabled"
      :readonly="readonly"
      @input="
        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
      "
    />
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  const props = defineProps({
    variant: {
      type: String as PropType<'primary'>,
      default: 'primary',
    },
    type: {
      type: String as PropType<'basic' | 'notepad'>,
      default: 'basic',
    },
    id: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
    modelValue: {
      type: String,
      required: true,
    },
    labelName: {
      type: String,
      required: true,
    },
    labelPosition: {
      type: String as PropType<'top' | 'left'>,
      default: 'top',
    },
  })

  defineEmits(['update:modelValue'])

  const classes = [
    `variant-${props.variant} label-position-${props.labelPosition}`,
    `${props.type === 'notepad' ? 'type-notepad' : ''}`,
  ] as Array<string>
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UITextarea.styles.scss';
</style>
