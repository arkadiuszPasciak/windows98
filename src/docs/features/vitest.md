# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
