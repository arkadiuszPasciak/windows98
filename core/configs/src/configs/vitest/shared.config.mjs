import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
