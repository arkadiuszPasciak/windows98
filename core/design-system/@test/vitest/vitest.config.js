import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
