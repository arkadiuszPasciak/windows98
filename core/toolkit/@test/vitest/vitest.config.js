import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
