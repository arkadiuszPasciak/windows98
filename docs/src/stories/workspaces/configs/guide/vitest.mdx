import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

