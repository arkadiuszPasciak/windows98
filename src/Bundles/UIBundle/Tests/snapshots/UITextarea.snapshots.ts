import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UITextarea from '@Bundles/UIBundle/Components/UITextarea.vue'

const leftPrimaryWrapper = mount(UITextarea, {
  props: {
    variant: 'primary',
    id: 'email',
    modelValue: 'your email...',
    labelName: 'Email',
    labelPosition: 'left',
  },
})

const topPrimaryWrapper = mount(UITextarea, {
  props: {
    variant: 'primary',
    id: 'address',
    modelValue: 'your address..',
    labelName: 'Address',
    labelPosition: 'top',
  },
})

describe('[UIBundle]<Snapshots>(UITextarea)', async () => {
  it('renders correctly component: leftPrimaryWrapper like as email', () => {
    expect(leftPrimaryWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: topPrimaryWrapper like as address', () => {
    expect(topPrimaryWrapper.element).toMatchSnapshot()
  })
})
