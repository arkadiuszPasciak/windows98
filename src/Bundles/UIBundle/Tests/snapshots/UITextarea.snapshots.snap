// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UITextarea) > renders correctly component: leftPrimaryWrapper like as email 1`] = `
<div
  class="UITextarea variant-primary label-position-left"
  data-v-cbf41378=""
>
  <label
    class="label"
    data-v-cbf41378=""
    for="email"
  >
    Email
  </label>
  <textarea
    class="textarea"
    data-v-cbf41378=""
    id="email"
  />
</div>
`;

exports[`[UIBundle]<Snapshots>(UITextarea) > renders correctly component: notepadWrapper like as notepad 1`] = `
<div
  class="UITextarea variant-primary label-position-top type-notepad"
  data-v-cbf41378=""
>
  <label
    class="label"
    data-v-cbf41378=""
    for="notepad-textarea"
  />
  <textarea
    class="textarea"
    data-v-cbf41378=""
    id="notepad-textarea"
  />
</div>
`;

exports[`[UIBundle]<Snapshots>(UITextarea) > renders correctly component: topPrimaryWrapper like as address 1`] = `
<div
  class="UITextarea variant-primary label-position-top"
  data-v-cbf41378=""
>
  <label
    class="label"
    data-v-cbf41378=""
    for="address"
  >
    Address
  </label>
  <textarea
    class="textarea"
    data-v-cbf41378=""
    id="address"
  />
</div>
`;
